documentation/docs/25-build-and-deploy/80-adapter-netlify.md

```diff
@@ -66,6 +66,48 @@ export default {
 };
 ```
 
+## Edge Middleware
+
+You can deploy one Netlify Edge Function [as middleware](https://docs.netlify.com/edge-functions/api/#modify-a-response) by placing an `edge-middleware.js` file at the root of your project. You can use it to intercept requests even for prerendered pages. Combined with using [server-side route resolution](configuration#router) you can make sure it runs prior to all navigations, no matter client- or server-side. This allows you to for example run A/B-tests on prerendered pages by rerouting a user to either variant A or B depending on a cookie.
+
+```js
+/// file: edge-middleware.js
+// @filename: ambient.d.ts
+declare module '@netlify/edge-functions';
+
+// @filename: index.js
+// ---cut---
+import { normalizeUrl } from '@sveltejs/kit';
+
+/**
+ * @param {Request} request
+ * @param {import('@netlify/edge-functions').Context} context
+ */
+export default async function middleware(request, { next, cookies }) {
+	const url = new URL(request.url);
+
+	if (url.pathname !== '/') return next();
+
+	// Retrieve feature flag from cookies
+	let flag = cookies.get('flag');
+
+	// Fall back to random value if this is a new visitor
+	flag ||= Math.random() > 0.5 ? 'a' : 'b';
+
+	// Set a cookie to remember the feature flags for this visitor
+	cookies.set('flag', flag);
+
+	// Get destination URL based on the feature flag
+	return new URL(flag === 'a' ? '/home-a' : '/home-b', url);
+}
+```
+
+By default middleware runs on all requests except for files within `_app/`. You can customize this by exporting a `export const config = { pattern: '<regex string>' }` object from the file similar to [how you can do it for native edge functions](https://docs.netlify.com/edge-functions/declarations/#declare-edge-functions-inline).
+
+> [!NOTE] Locally during dev and preview this only approximates the capabilities of edge functions. Notably, you cannot read the request or response body, and many properties on the context object are `null`ed.
+
+> [!NOTE] If you want to run code prior to a request but neither have prerendered pages nor rerouting logic, then it makes more sense to use the [handle hook](hooks#Server-hooks-handle) instead.
```

Comment:
```suggestion
> [!NOTE] If you can do what you need to by using the [handle hook](hooks#Server-hooks-handle), do so. Edge middleware has a not-insignificant performance penalty and should only be used if absolutely necessary. Notable use cases include A/B testing using rerouting and static assets, or adding headers to requests for static assets.
```

Just feels like putting the "do not use unless you need it" warning first is a good idea.